erts/emulator/test/bs_construct_SUITE.erl

```diff
@@ -889,6 +890,33 @@ bs_append_overflow_unsigned() ->
     C = <<A/binary,1,B/binary>>,
     true = byte_size(B) < byte_size(C).
 
+reductions(_Config) ->
+    TwoMeg = <<0:(2_000*1024)/unit:8>>,
+    reds_at_least(2000, fun() -> <<0:8,TwoMeg/binary>> end),
+    reds_at_least(4000, fun() -> <<0:8,TwoMeg/binary,TwoMeg/binary>> end),
+    reds_at_least(1000, fun() -> <<0:8,TwoMeg:(1000*1024)/binary>> end),
+
+    %% Here we expect about 500 reductions in the bs_append
+    %% instruction for setting up a writable binary and about 2000
+    %% reductions in the bs_put_binary instruction for copying the
+    %% binary data.
+    reds_at_least(2500, fun() -> <<TwoMeg/binary,TwoMeg:(2000*1024)/binary>> end),
+    ok.
+
+reds_at_least(N, Fun) ->
+    receive after 1 -> ok end,
+    {reductions,Red0} = process_info(self(), reductions),
+    _ = Fun(),
+    {reductions,Red1} = process_info(self(), reductions),
+    Diff = Red1 - Red0,
+    io:format("Expected at least ~p; got ~p\n", [N,Diff]),
```

Comment:
a question: is it ok allowing console output in OTP test runs?

Reply:
The output from `io:format/2` will go to the log file for the test case, not to the console.